ompi/mca/mtl/ofi/mtl_ofi_component.c

```diff
@@ -802,14 +791,12 @@ ompi_mtl_ofi_component_init(bool enable_progress_threads,
     opal_argv_free(exclude_list);
     exclude_list = NULL;
 
-#if OPAL_CUDA_SUPPORT
     if (!(prov->caps & FI_HMEM)) {
         opal_output_verbose(1, opal_common_ofi.output,
```

Comment:
This printf is now going to always fire for providers that don't support FI_HMEM.  That's a fairly big change in behavior.  More thought is needed here.

Reply:
This note still applies.

Reply:
Changing behavior to:

Try to select a provider with FI_HMEM support. If this fails, proceed selecting provider w/o FI_HMEM support anyway but set a bool that declares accelerator transfers aren't supported. Change MTL components to report back if they support directly sending using accelerator buffers. If not, copy from accelerator to host buffers.